src/blockchain/electrum.rs

```diff
@@ -325,3 +327,29 @@ crate::bdk_blockchain_tests! {
         ElectrumBlockchain::from(Client::new(&test_client.electrsd.electrum_url).unwrap())
     }
 }
+
+#[cfg(test)]
+#[cfg(feature = "test-electrum")]
+mod test {
+    use std::sync::Arc;
+
+    use super::*;
+    use crate::testutils::blockchain_tests::TestClient;
+
+    #[test]
+    fn test_electrum_blockchain_factory() {
+        let test_client = TestClient::default();
+
+        let factory = Arc::new(ElectrumBlockchain::from(
+            Client::new(&test_client.electrsd.electrum_url).unwrap(),
+        ));
+
+        let a = factory.build("aaaaaa", None).unwrap();
+        let b = factory.build("bbbbbb", None).unwrap();
+
+        assert_eq!(
+            a.client.block_headers_subscribe_raw().unwrap().height,
+            b.client.block_headers_subscribe().unwrap().height
```

Comment:
Instead of creating an `Arc::new()` does it make sense to have an api like `BlockchainFactory::from(T : Blockchain)` which returns and `Arc<Blockchain>`.

And then the trait can have a function like `BlockchainFactory::build_for_wallet` which returns the arc clone of the inner blockchain.

So the instantiation would look like
` let a = factory.build_for_wallet("aaaaa", None).unwrap()`.

To me it seems more clear that way for lib users to intuitively understand whats happening..

Reply:
I think it would be a bit confusing because it would only work for `StatelessBlockchain`. For the others you'd still need to create it manually, like for rpc

Reply:
To clear my understanding further, what does it mean exactly by `StatelessBlockchain`?? Like a blockchain structure that doesn't contain or store and wallet level data?? And is interoperable with multiple wallets connecting to it?? 

So in that sense `electrum` and `esplora` are stateful, because they store wallet specific data in them??